lib/src/validators/collection_validators.dart

```diff
@@ -0,0 +1,311 @@
+/// `FormBuilderLocalizations.current.maxLengthErrorText(max)`.
+///
+/// ## Type Parameters
+/// - `T`: The type of input to validate. Must extend `Object` and can be a
+///   `String`, `Iterable`, `Map`, or other object type.
+///
+/// ## Parameters
+/// - `max` (`int`): The maximum length allowed. Must be non-negative.
+/// - `maxLengthMsg` (`String Function(T input, int max)?`): Optional
+///   function to generate custom error messages. Receives the input and the
+///   maximum length allowed and returns an error message string.
+///
+/// ## Returns
+/// Returns a `Validator<T>` function that takes an input of type T and returns:
+/// - `null` if the input's length is less than or equal to [max]
+/// - An error message string if the validation fails
+///
+/// ## Throws
+/// - `AssertionError`: When [max] is negative
+///
+/// ## Examples
+/// ```dart
+/// // String validation
+/// final stringValidator = maxLength<String>(5);
+/// print(stringValidator('hello')); // Returns null
+/// print(stringValidator('too long')); // Returns error message
+///
+/// // List validation
+/// final listValidator = maxLength<List>(3);
+/// print(listValidator([1, 2])); // Returns null
+/// print(listValidator([1, 2, 3, 4])); // Returns error message
+///
+/// // Custom error message
+/// final customValidator = maxLength<String>(
+///   3,
+///   maxLengthMsg: (_, max) => 'Text cannot exceed $max characters!',
+/// );
+/// ```
+///
+/// ## Caveats
+/// - Non-collection objects (those that are neither String, Iterable, nor Map)
+///   are treated as having a length of 1
+/// {@endtemplate}
+Validator<T> maxLength<T extends Object>(int max,
+    {String Function(T input, int max)? maxLengthMsg}) {
+  assert(max >= 0, 'The "maxLength" parameter may not be negative');
+  return (T input) {
+    // I think it makes more sense to say that scalar objects has length 1 instead of 0.
+    int valueLength = 1;
+
+    if (input is String) {
+      valueLength = input.length;
+    } else if (input is Iterable) {
+      valueLength = input.length;
+    } else if (input is Map) {
+      valueLength = input.length;
+    }
+
+    return valueLength > max
+        ? maxLengthMsg?.call(input, max) ??
+            FormBuilderLocalizations.current.maxLengthErrorText(max)
+        : null;
+  };
+}
+
+/// {@template validator_between_length}
+/// Creates a validator function that checks if the input's length falls within
+/// an inclusive range between `min` and `max`. The validator returns `null` for
+/// valid input and an error message for invalid input.
+///
+/// If validation fails and no custom error message generator is provided via
+/// [betweenLengthMsg], returns the default localized error message from
+/// `FormBuilderLocalizations.current.betweenLengthErrorText(min, max)`.
+///
+/// ## Type Parameters
+/// - `T`: The type of input to validate. Must extend `Object` and can be a
+///   `String`, `Iterable`, `Map`, or other object type.
```

Comment:
Could be another object type? Make sense this?
If validator recieve for example an int, never will validate a `valueLength` of object.